client/src/EventListener/UnauthorizedExceptionListener.php

```diff
@@ -11,11 +11,14 @@ class UnauthorizedExceptionListener
 {
     public function onKernelException(ExceptionEvent $exceptionEvent)
     {
+        $request = $exceptionEvent->getRequest();
         $exception = $exceptionEvent->getThrowable();
 
         if ($exception instanceof HttpException) {
             if (Response::HTTP_UNAUTHORIZED == $exception->getStatusCode()) {
-                $response = new RedirectResponse('/login?from=api');
+                $url = '/login?from=api&lastPage=' . urlencode($request->getRequestUri());
```

Comment:
Additional thought. Can you craft a request url to Digideps that throws an unauthorised and injects arbitrary values here that are passed down the chain?

Reply:
I've tried to pass random values to it, trying to craft path traversal, offsite, other user valid URL's & special characters. They are all passed back into the URL but when it meets the validator it's all correctly ignored.